Application/src/main/java/com/example/android/mediarecorder/MainActivity.java

```diff
@@ -206,6 +226,50 @@ private boolean prepareVideoRecorder(){
         return true;
     }
 
+    private boolean areCameraPermissionGranted() {
+
+        for (String permission : requiredPermissions){
+            if (!(ActivityCompat.checkSelfPermission(this, permission) ==
+                    PackageManager.PERMISSION_GRANTED)){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void requestCameraPermissions(){
+        ActivityCompat.requestPermissions(
+                this,
+                requiredPermissions,
+                MEDIA_RECORDER_REQUEST);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+
+        if (MEDIA_RECORDER_REQUEST != requestCode) {
+            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+            return;
+        }
+
+        boolean areAllPermissionsGranted = true;
+        for (int result : grantResults){
+            if (result != PackageManager.PERMISSION_GRANTED) areAllPermissionsGranted = false;
```

Comment:
you break after the `areAllPermissionsGranted = false` so you don't need to iterate through all, if one is false.